tests/fast_int.cpp

```diff
@@ -0,0 +1,379 @@
+out of range base tests - should still work even with a base greater than 36
+*/
+
+int main()
+{
+
+  // int basic test
+  const std::vector<int> int_basic_test_expected { 0, 10, -40, 1001, 9 };
+  const std::vector<std::string> int_basic_test { "0", "10 ", "-40", "1001 with text", "9.999" };
+
+  for (std::size_t i = 0; i < int_basic_test.size(); ++i)
+  {
+    const auto& f = int_basic_test[i];
+    int result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result);
+    if (answer.ec != std::errc()) {
+      std::cerr << "could not convert to int for input: " << std::quoted(f) << std::endl;
+      return EXIT_FAILURE;
+    }
+    else if (result != int_basic_test_expected[i]) {
+      std::cerr << "result "  << std::quoted(f) << " did not match with expected int: " << int_basic_test_expected[i] << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // unsigned basic test
+  const std::vector<unsigned> unsigned_basic_test_expected { 0, 10, 1001, 9 };
+  const std::vector<std::string> unsigned_basic_test { "0", "10 ", "1001 with text", "9.999" };
+
+  for (std::size_t i = 0; i < unsigned_basic_test.size(); ++i)
+  {
+    const auto& f = unsigned_basic_test[i];
+    unsigned result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result);
+    if (answer.ec != std::errc()) {
+      std::cerr << "could not convert to unsigned for input: " << std::quoted(f) << std::endl;
+      return EXIT_FAILURE;
+    }
+    else if (result != unsigned_basic_test_expected[i]) {
+      std::cerr << "result "  << std::quoted(f) << " did not match with expected unsigned: " << unsigned_basic_test_expected[i] << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // int invalid error test
+  const std::vector<std::string> int_invalid_argument_test{ "text", "text with 1002", "+50", " 50" };
+
+  for (std::size_t i = 0; i < int_invalid_argument_test.size(); ++i)
+  {
+    const auto& f = int_invalid_argument_test[i];
+    int result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result);
+    if (answer.ec != std::errc::invalid_argument) {
+      std::cerr << "expected error should be 'invalid_argument' for: " << std::quoted(f) << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // unsigned invalid error test
+  const std::vector<std::string> unsigned_invalid_argument_test{ "text", "text with 1002", "+50", " 50", "-50" };
+
+  for (std::size_t i = 0; i < unsigned_invalid_argument_test.size(); ++i)
+  {
+    const auto& f = unsigned_invalid_argument_test[i];
+    unsigned result;
+    auto answer = fast_float::from_chars(f.data(), f.data() + f.size(), result);
+    if (answer.ec != std::errc::invalid_argument) {
+      std::cerr << "expected error should be 'invalid_argument' for: " << std::quoted(f) << std::endl;
+      return EXIT_FAILURE;
+    }
+  }
+
+  // int out of range error test 
+  const std::vector<std::string> int_out_of_range_test{ "2000000000000000000000" };
```

Comment:
add some more tests here, especially when the numbers are just outside range (like int_max + 1 or int_min -1). these are a bit tougher to catch